app/codes/consensus/consensus.py

```diff
@@ -77,55 +83,115 @@ def get_node_trust_score(public_key):
 #     return True
 
 
-def check_community_consensus(block):
+def get_committee_consensus(block):
+    """
+    Check a block for consensus
+    Return -1 for invalid block conensus
+    Return 1 for valid block conensus
+    Return 0 for no conensus
+    """
     receipts_in_temp = get_receipts_from_storage(block['index'])
 
     for receipt in receipts_in_temp:
         append_receipt_to_block(block, receipt)
 
-    receipt_counts = validate_block_receipts(block)
+    receipt_counts = count_block_receipts(block)
     committee = block['data']['committee']
 
-    # TODO - Deal with the case when minimum number of committee members are not avalable
-    # if len(committee) < 3:
-    #    if receipt_counts['positive_receipt_count'] > 0:
-    #        return True
-    # Block is received from sentinel node after a timeout.
     if len(committee) == 1:
         if committee[0] == Configuration.config("SENTINEL_NODE_WALLET"):  # Todo - Check if block is empty
             logger.info('Consensus satisfied as sentinel node is the miner')
-            return True
+            return 1
         else:
-            return False
+            return 0
     if len(committee) < MINIMUM_ACCEPTANCE_VOTES:
-        return False
+        return 0
 
     if receipt_counts['positive_receipt_count'] + 1 >= MINIMUM_ACCEPTANCE_VOTES:
         # TODO - Check if time elapsed has exceeded receipt cut off. Do not accept otherwise
         # This is to give every node sometime to send their receipts for the block
-        return True
-    return False
+        return 1
+    elif receipt_counts['negative_receipt_count'] >= MINIMUM_ACCEPTANCE_VOTES:
+        return -1
+    return 0
 
 
-def validate_block_miner(block):
-    miner_address = block['creator_wallet']
+def validate_block_miner_committee(block):
+    creator_wallet = block['data']['creator_wallet']
 
     expected_miner = get_miner_for_current_block()['wallet_address']
 
     if expected_miner is None:
         return True
 
-    if miner_address != expected_miner:
-        logger.info(f"Invalid miner {miner_address} for block. Expected {expected_miner}")
+    if creator_wallet != expected_miner:
+        logger.info(f"Invalid miner {creator_wallet} for block. Expected {expected_miner}")
         return False
+    
+    expected_commitee = get_committee_wallet_list_for_current_block()
+    if set(block['data']['committee']) != set(expected_commitee):
+        logger.info(f"Invalid committee {block['data']['committee']} for block. Expected {expected_commitee}")
+        return False
+    
+    # Signature checks
+    # TODO - Check if block hash is being validated anywhere else. If not add it here. 
+    miner_vote_found = False
+    for receipt in block['receipts']:
+        if not validate_receipt(receipt):
```

Comment:
@kousthubraja we should validate only the committee members' receipts. It is possible that a malicious committee member appends a false receipt but the rest of the committee is able to reach consensus with the valid ones.

Reply:
it's checked [here](https://github.com/newrlfoundation/newrl/blob/9aa6e26ad51ffdd8413171b2552074e3070e506c/app/codes/consensus/consensus.py#L145). the entire block will be rejected as no honest miner would append a non-committee receipt to the block.